packages/admin/cms-admin/src/generator/future/generator.ts

```diff
@@ -30,18 +30,27 @@ export function isFormFieldConfig<T>(arg: any): arg is FormFieldConfig<T> {
     return !isFormLayoutConfig(arg);
 }
 
-export type FormLayoutConfig<T> = {
-    type: "fieldSet";
-    name: string;
-    title: string;
-    supportText?: string; // can contain field-placeholder
-    collapsible: boolean; // default true
-    initiallyExpanded: boolean; // default false
-    fields: FormFieldConfig<T>[];
+type OptionalFieldsConfig<T> = {
+    type: "optionalFields";
+    name: keyof T; // object name containing fields
+    checkboxLabel: string;
```

Comment:
the default value could be auto-generated from name

(same for fieldSet title)

Reply:
added feature for fieldSet: https://github.com/vivid-planet/comet/pull/2379